lib/tasks/BackbeatTask.js

```diff
@@ -42,30 +23,32 @@ class BackbeatTask {
                 }
 
                 const now = Date.now();
-                if (now > (startTime + self.config.retryTimeoutS * 1000)) {
-                    log.error(
-                        `retried for too long to ${actionDesc}, giving up`,
+                const retriesExceeded = nbRetries > this.retryParams.maxRetries;
+                const timeoutReached = now >
+                    (startTime + this.retryParams.timeoutS * 1000);
+                if (retriesExceeded || timeoutReached) {
+                    log.error('giving up replication as retries exceeded',
```

Comment:
`actionDesc` should be used for the log instead of "replication", also I would rephrase "retries exceeded" to "giving up retries" to be agnostic as to why we stopped retrying (the reason can be found as `retriesExceeded` or `timeoutReached` in the log params).